lib/galaxy/tool_util/verify/asserts/text.py

```diff
@@ -1,10 +1,14 @@
 import re
 
 
-def assert_has_text(output, text):
+def assert_has_text(output, text, n=None):
     """ Asserts specified output contains the substring specified by
     the argument text."""
-    assert output.find(text) >= 0, "Output file did not contain expected text '%s' (output '%s')" % (text, output)
+    if not n:
```

Comment:
In the unlikely case that the function is called with `n=0`, this clause will execute, so we assert that `text` exists, although the argument implies 0 occurrences (`n=0`), which is not the expected behavior, I think. Maybe we should add a comment that if n is not set or n=0, the assertion will be made for 1 or more occurrences? Or, if *there is* a use case for checking for 0 occurrences (I don't know if there is), we could handle it here (if n is explicitly set to 0).

Reply:
I think `if n is not None` (`is not`, not `!=`) may be a better option (as per @simonbray 's suggestion). `n=1` implies a count of **1**. So we would be changing assumptions for existing code (code that calls this function without the `n` argument and assumes 1 *or more* occurrences of `text` - we'd be changing that to exactly 1. The most obvious default, I think, would be "1 or more", with "not set" being the next best thing.

Reply:
You're right, I didn't think that through.